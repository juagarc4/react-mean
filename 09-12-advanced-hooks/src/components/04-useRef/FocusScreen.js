import React, { useRef } from 'react'

export const FocusScreen = () => {
  const inputRef = useRef()

  const handleClick = () => {
    inputRef.current.select()
  }
  return (
    <div>
      <h1>useRef</h1>
      <hr />
      <input
        ref={inputRef}
        className="form-control"
        placeholder="Your name"
      ></input>
      <button className="btn btn-outline-primary mt-3" onClick={handleClick}>
        Focus
      </button>
    </div>
  )
}
