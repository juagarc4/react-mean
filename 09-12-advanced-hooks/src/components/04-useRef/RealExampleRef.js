import React, { useState } from 'react'
import { MultipleCustomHooks } from 'components/03-examples/MultipleCustomHooks'

export const RealExampleRef = () => {
  const [show, setShow] = useState(false)
  return (
    <div>
      <h1>Real Example of useRef</h1>
      <hr />

      {show && <MultipleCustomHooks />}
      <button
        className="btn btn-outline-primary mt-3"
        onClick={() => setShow(!show)}
      >
        Show/Hide
      </button>
    </div>
  )
}
