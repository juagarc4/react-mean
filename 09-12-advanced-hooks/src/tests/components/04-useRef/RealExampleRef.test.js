import React from 'react'
import { shallow } from 'enzyme'
import { RealExampleRef } from 'components/04-useRef/RealExampleRef'

describe('Tests in <RealExampleRef />', () => {
  const wrapper = shallow(<RealExampleRef />)

  test('Should show correctly', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false)
  })
  test('Should show the component <MultipleCustomHooks />', () => {
    wrapper.find('button').simulate('click')
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true)
  })
})
