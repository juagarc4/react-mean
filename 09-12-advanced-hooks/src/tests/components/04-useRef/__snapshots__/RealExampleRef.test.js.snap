// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <RealExampleRef /> Should show correctly 1`] = `
<div>
  <h1>
    Real Example of useRef
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary mt-3"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
